### PR TITLE
[IMP] account: making account_get in analytic defaults inheritable

### PR DESCRIPTION
**Pull request by hbto**
Main
-

[IMP] account: making account_get in analytic defaults inheritable

Explanation
-

It is too hard to try inherit on the method `account_get` on `account.analytic.default` without overriding it in full.

By means of the new compaction of code the model `account.analytic.default` will be more easily extended.

Regards.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hbto** [comment]
cc/ @moylop260

File: addons/account/models/account_analytic_default.py [hunk 1]
```diff
@@ -27,8 +27,7 @@ def _check_account_or_tags(self):
         if any(not default.analytic_id and not default.analytic_tag_ids for default in self):
             raise ValidationError(_('An analytic default requires at least an analytic account or an analytic tag.'))
 
-    @api.model
-    def account_get(self, product_id=None, partner_id=None, account_id=None, user_id=None, date=None, company_id=None):
+    def _retrieve_account_get_domain(self, product_id=None, partner_id=None, account_id=None, user_id=None, date=None, company_id=None):
```
Comment (moylop260) [inline addons/account/models/account_analytic_default.py:30]:
```suggestion
    @api.model
    def _retrieve_account_get_domain(self, product_id=None, partner_id=None, account_id=None, user_id=None, date=None, company_id=None):
```

File: addons/account/models/account_analytic_default.py [hunk 2]
```diff
@@ -48,17 +47,27 @@ def account_get(self, product_id=None, partner_id=None, account_id=None, user_id
         if date:
             domain += ['|', ('date_start', '<=', date), ('date_start', '=', False)]
             domain += ['|', ('date_stop', '>=', date), ('date_stop', '=', False)]
+        return domain
+
+    def _retrieve_account_get_index(self):
+        self.ensure_one()
+        index = 0
+        if self.product_id: index += 1
```
Comment (moylop260) [inline addons/account/models/account_analytic_default.py:55]: Fix lint

```suggestion
        if self.product_id:
            index += 1
```

**Review by moylop260 — APPROVED** [review]
👍  LGTM

WDYT @jbw-odoo @qdp-odoo

**Comment by C3POdoo** [comment]
Dear @hbto,

Thank you for your contribution but the version 14.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**